### PR TITLE
Button: Stabilize `__experimentalIsFocusable` prop

### PR DESCRIPTION
**Pull request by mirka**
## What?

Stabilizes the `__experimentalIsFocusable` prop on the Button component into a `accessibleWhenDisabled` prop.

### TODO

- [x] Fix eslintrc cascade around `no-restricted-syntax` rule (#62306)
- [x] Migrate existing usage in repo (#63107)
- [x] Adjust eslint rule for new prop (#63107)

## Why?

We are tightening our standards (#62080) around disabled Button accessibility.

The `accessibleWhenDisabled` naming and pattern aligns with what Ariakit is doing, which gives some confidence that this is sustainable and scalable across a component library.

## How?

- Improve documentation to encourage usage and developer education.
- [Passive deprecate](https://github.com/WordPress/gutenberg/issues/61099) the `__experimentalIsFocusable` prop.

## Testing Instructions

- Readme and Storybook props table descriptions should look good.
- Static checks on CI (i.e. the linter) should pass after #63107 is merged into this branch.

---

## ✍️ Dev Note

The `Button` component had a `__experimentalIsFocusable` prop to keep the button focusable when disabled. This prop is now stabilized and renamed to `accessibleWhenDisabled`. (Existing usages of `__experimentalIsFocusable` will continue to work indefinitely.)

In most cases, it is recommended to set this to `true`. Disabling a control without maintaining focusability can cause accessibility issues, by hiding their presence from screen reader users, or by preventing focus from returning to a trigger element.

Learn more about the [focusability of disabled controls](https://www.w3.org/WAI/ARIA/apg/practices/keyboard-interface/#focusabilityofdisabledcontrols) in the WAI-ARIA Authoring Practices Guide.

### CODE REVIEW

**Comment by stokesman** [comment]
I'd kind of wish `disabled` could be made `boolean|'accessibly'` rather than adding a new prop. That’s just the first thought I had and maybe it’s a nonstarter for some reason. Good to see this anyway 👍.

**Comment by mirka** [comment]
> I'd kind of wish `disabled` could be made `boolean|'accessibly'` rather than adding a new prop. That’s just the first thought I had and maybe it’s a nonstarter for some reason.

It did [come up](https://github.com/WordPress/gutenberg/issues/56547#issuecomment-1833753814), with the two major downsides being breaking spec of an extremely standard prop/attribute, and having to use a mixed type. I wouldn't go as far as to say it's a nonstarter, but breaking spec is a big one for me, personally.

To add some context about going the "additional prop" route, I added a note in the PR description about how Ariakit adopts this pattern.

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,7 @@
 ### Enhancements
 
 -   `ToolbarButton`: Deprecate `isDisabled` prop and merge with `disabled` ([#63101](https://github.com/WordPress/gutenberg/pull/63101)).
+-   `Button`: Stabilize `__experimentalIsFocusable` prop ([#62282](https://github.com/WordPress/gutenberg/pull/62282)).
```
Comment (tyxla) [inline packages/components/CHANGELOG.md]: Is the stable prop name worth a mention in the CHANGELOG? Could be useful if someone is researching when that prop was introduced.

Perhaps like this:

```suggestion
-   `Button`: Stabilize `__experimentalIsFocusable` prop as `accessibleWhenDisabled` ([#62282](https://github.com/WordPress/gutenberg/pull/62282)).
```

File: packages/components/src/button/index.tsx [hunk 2]
```diff
@@ -159,7 +161,7 @@ export function UnforwardedButton(
 		'has-icon': !! icon,
 	} );
 
-	const trulyDisabled = disabled && ! isFocusable;
+	const trulyDisabled = disabled && ! accessibleWhenDisabled;
```
Comment (tyxla) [inline packages/components/src/button/index.tsx:164]: Nice to see this getting clearer now 👏

File: packages/components/src/button/test/index.tsx [hunk 3]
```diff
@@ -619,6 +619,14 @@ describe( 'Button', () => {
 				'mixed'
 			);
 		} );
+
+		it( 'should not break when the legacy __experimentalIsFocusable prop is passed', () => {
```
Comment (tyxla) [inline packages/components/src/button/test/index.tsx:623]: 👍 Thanks for adding a test!

**Review by tyxla — APPROVED** [review]
Nice work, thanks @mirka 👍 🚀 

LGTM and tests well, together with the accompanying PR.

**Pull request merged**